src/smooth move animations.js

```diff
@@ -0,0 +1,109 @@
+/**
+👣
+@file smooth move animations
+@summary Define unique walking and idle tiles in each direction for any event.
+@license MIT
+@author Violgamba (Jon Heard)
+
+
+@description
+Builds on the smooth-move plugin to let you define unique walking and idle tiles in each direction
+for any event.  Useful if using bipsi for a more graphically involved game.
+
+NOTE - This plugin requires the smooth-move plugin in order to work.  It also needs to come AFTER
+smooth-move, so set it's "plugin-order" to something higher than what the smooth-move plugin's
+plugin-order is set to.
+
+This works by letting you define extra tile type fields for different movement states of an event.
+None of the extra tile fields are required.  Any missing fields are replaced with logical defaults.
+All direction tiles default to the generic "graphic-idle" and "graphic-move" tiles.  Meanwhile, the
+"graphic-idle" and "graphic-move" tiles default to the base "graphic" tile.  And if the event has
+no "graphic" tile field then the event just won't be visible, as normal.
+
+
+Here is the full list of the tile fields smooth-move-animations can use:
+
+SMOOTH-MOVE-ANIMATIONS TILE FIELD KEYS:
+graphic-idle, graphic-move,
```

Comment:
noticed a minor bug here: when `graphic-idle` is configured and the game is started, the player will use their regular `graphic` field until after they have moved once

Reply:
The problem is that the animation plugin does lazy setup for events.  This is triggered the first time an event moves, be that the avatar or a walking event (no distinction is made).  The pro is that only events that actually move are setup for movement.  The con is that they aren't setup until their first move.

The solution I landed on was to manually set each moving event's _regular_ `graphic` field to whatever tile it should start with before first movement.  I should probably document this as an official part of the plugin's setup process.  An alternative half-solution would be to have the animation plugin explicitly setup the avatar's animation.  It's only a half-solution since it doesn't fix the problem for non-avatar moving events.

Can you think of any better solutions?  I _could_ require an explicit tag for events that should be auto-setup by the animator, but that seems cumbersome.

Reply:
thinking on this some more, the best solution here may actually be much simpler than expected: you could just remove `graphic-idle`. if i understand correctly, `graphic-idle` is redundant with the standard `graphic` field aside from this lazy setup issue, and it's only the move + directional idle fields that actually provide new functionality.

Reply:
This is a good idea.  'graphic' would need to replace both 'graphic-idle' and 'graphic-idle-down'.  The naming inconsistency is unfortunate.  However, given that the alternative is a field who's definition is meaningful only between game start and a moving event's first move, this seems reasonable.

Also, I am thinking that the documentation might not be explicit enough on one point.  You'd generally either use 'graphic-idle' and 'graphic-move' OR you'd use 'graphic-idle-down', 'graphic-idle-up', ... 'graphic-move-left', 'graphic-move-right'.  The directions are the most detailed setup and the idle/move is a simper alternative.  I'll update the documentation to make that more explicit.